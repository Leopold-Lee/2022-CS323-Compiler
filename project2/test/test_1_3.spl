int test_1_3(int a, int b)
{
  int c = b || a && b;
  c = a == b;
  if (a > b)
  {
    return a;
  }
  else if (b == a){
    return b;
  } else {
    return b;
  }
}
