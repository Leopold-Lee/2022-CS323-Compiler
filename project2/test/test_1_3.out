Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: test_1_3
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: a
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: b
        RP
      CompSt (2)
        LC
        DefList (3)
          Def (3)
            Specifier (3)
              TYPE: int
            DecList (3)
              Dec (3)
                VarDec (3)
                  ID: c
                ASSIGN
                Exp (3)
                  Exp (3)
                    ID: b
                  OR
                  Exp (3)
                    Exp (3)
                      ID: a
                    AND
                    Exp (3)
                      ID: b
            SEMI
        StmtList (4)
          Stmt (4)
            Exp (4)
              Exp (4)
                ID: c
              ASSIGN
              Exp (4)
                Exp (4)
                  ID: a
                EQ
                Exp (4)
                  ID: b
            SEMI
          StmtList (5)
            Stmt (5)
              IF
              LP
              Exp (5)
                Exp (5)
                  ID: a
                GT
                Exp (5)
                  ID: b
              RP
              Stmt (6)
                CompSt (6)
                  LC
                  StmtList (7)
                    Stmt (7)
                      RETURN
                      Exp (7)
                        ID: a
                      SEMI
                  RC
              ELSE
              Stmt (9)
                IF
                LP
                Exp (9)
                  Exp (9)
                    ID: b
                  EQ
                  Exp (9)
                    ID: a
                RP
                Stmt (9)
                  CompSt (9)
                    LC
                    StmtList (10)
                      Stmt (10)
                        RETURN
                        Exp (10)
                          ID: b
                        SEMI
                    RC
                ELSE
                Stmt (11)
                  CompSt (11)
                    LC
                    StmtList (12)
                      Stmt (12)
                        RETURN
                        Exp (12)
                          ID: b
                        SEMI
                    RC
        RC
